launch_ros/launch_ros/actions/node.py

```diff
@@ -385,11 +379,21 @@ def _perform_substitutions(self, context: LaunchContext) -> None:
         if self.__expanded_node_namespace != '/':
             self.__final_node_name += self.__expanded_node_namespace
         self.__final_node_name += '/' + self.__expanded_node_name
+        # expand global parameters first,
+        # so they can be overriden with specific parameters of this Node
+        global_params = context.launch_configurations.get('ros_params', None)
+        if global_params is not None or self.__parameters is not None:
+            self.__expanded_parameter_files = []
+        if global_params is not None:
+            param_file_path = self._create_params_file_from_dict(global_params)
+            self.__expanded_parameter_files.append(param_file_path)
+            cmd_extension = ['--params-file', f'{param_file_path}']
+            self.cmd.extend([normalize_to_list_of_substitutions(x) for x in cmd_extension])
+            assert os.path.isfile(param_file_path)
```

Comment:
@ivanpauno here we `assert`, but below we may ignore a missing file under the same circumstances. I think we should be consistent.

Reply:
it's not exactly the same, this is a temporal file created a few lines before.

The other might be a temporal file or a file passed by the user.
I can add an assert for the temp file, but it's not the goal of this PR to change what happens when the user passes a wrong path.

Reply:
done https://github.com/ros2/launch_ros/pull/158/commits/f66375545bf968f686c60e94fbc961f853b5109c